core/vibes/soul/primitives/banner/index.tsx

```diff
@@ -0,0 +1,69 @@
+'use client';
+
+import { clsx } from 'clsx';
+import { X } from 'lucide-react';
+import { ForwardedRef, forwardRef, ReactNode, useCallback, useEffect, useState } from 'react';
+
+export const Banner = forwardRef(
+  (
+    {
+      children,
+      hideDismiss = false,
+      className,
+      onDismiss,
+    }: {
+      children: ReactNode;
+      hideDismiss?: boolean;
+      className?: string;
+      onDismiss?(): void;
+    },
+    ref: ForwardedRef<HTMLDivElement>,
+  ) => {
+    const [banner, setBanner] = useState({ dismissed: false, initialized: false });
+
+    useEffect(() => {
+      const hidden = localStorage.getItem('hidden-banner') === 'true';
```

Comment:
I think right now there's a pretty bad layout shift because we're using local storage for banners. I think we need to move to using cookies for this. We can either remove banner support for this PR and make a separate issue for it to track progress on that or we could update VIBES to fix the banner before we merge this.